website/components/PropTable.tsx

```diff
@@ -0,0 +1,195 @@
+import React from 'react';
+import { css } from 'emotion';
+import { Table, TableHeader, Row, Cell } from '@leafygreen-ui/table';
+import { Subtitle, InlineCode } from '@leafygreen-ui/typography/dist';
+import PropDefinition from 'components/PropDefinition';
+import TypographyPropTable from 'components/TypographyPropTable';
+
+const subtitleBottomMargin = css`
+  margin-bottom: 24px;
+`;
+
+const tableBottomMargin = css`
+  margin-bottom: 56px;
+`;
+
+/**
+ * Syntactic units in unist syntax trees are called nodes.
+ * Interface dervied from Unified types
+ */
+export interface Node {
+  /**
+   * The variant of a node.
+   */
+  type: string;
+
+  /**
+   * Information from the ecosystem.
+   */
+  data?: {
+    [key: string]: unknown;
+  };
+
+  /**
+   * Location of a node in a source document.
+   * Must not be present if a node is generated.
+   */
+  position?: Position;
+
+  children?: Array<any>;
+
+  [key: string]: unknown;
+}
+
+const map = {
+  [0]: 'prop',
+  [1]: 'type',
+  [2]: 'description',
+  [3]: 'default',
+} as const;
+
+interface TableDataInterface {
+  prop: string;
+  type: string;
+  description: string;
+  default: string;
+}
+
+function getTableData(rows: Array<any>): Array<TableDataInterface> {
+  if (!rows) {
+    return null;
+  }
+
+  const rowMap = rows.map(row => {
+    const rowObj: Partial<TableDataInterface> = {};
+    row.children.map((cell, index) => {
+      const value =
+        cell.children
+          .map(child => {
+            if (child.children) {
+              return child.children.map(child => child.value).join('');
+            }
+
+            return child.value;
+          })
+          .join('') || '-';
+
+      rowObj[map[index]] = value;
+    });
+    return rowObj;
+  });
+
+  rowMap.shift();
+
+  return rowMap as Array<TableDataInterface>;
+}
+
+function PropTable({ mdAst, component }: { mdAst: Node; component: string }) {
```

Comment:
could use a more descriptive variable name, e.g. `markdownAst`

Reply:
Given that the structure of the readme files are known ahead of time, why can't we give this an exact type?

Reply:
`mdAst` is typed as `Node` but we're only using the `children` field, so most of these fields are unused. Also `mdAst.children` is typed as `Array<any>`, but `getTableData` is definitely making more assumptions about the structure of the data. These assumptions would be more clear if they're typed.

Reply:
Updated

Reply:
don't see a new push?

Reply:
🤦 it failed the first time and I didn't notice. Updated for real this time